### PR TITLE
Add UM-70

### PR DESCRIPTION
**Pull request by duoshock**
## Description
Please add UM-70 keyboard. UM-70 is a 70 keys qwerty split keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/merge/um70/um70.h [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "quantum.h"
+
+#define XXX KC_NO
+
+#define LAYOUT_RSpace_SplitBksp( \
```
Comment (drashna) [inline keyboards/merge/um70/um70.h:23]:
```suggestion
#define LAYOUT_rspace_split_bksp( \
```

File: keyboards/merge/um70/um70.h [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    { L30, L31, L32, L33, L34, L35, XXX, XXX }, \
+    { L40, L41, L42, L43, L44, L45, L46, L47 }, \
+    { R00, R01, R02, R03, R04, R05, R06, R07 }, \
+    { R10, R11, R12, R13, R14, R15, R16, R17 }, \
+    { R20, R21, R22, R23, R24, R25, R26, R27 }, \
+    { R30, R31, R32, R33, R34, R35, R36, R37 }, \
+    { R40, R41, R42, R43, R44, R45, XXX, R47 } \
+}
+
+#define LAYOUT_RSpace_2UBksp( \
```
Comment (drashna) [inline keyboards/merge/um70/um70.h:42]:
```suggestion
#define LAYOUT_rspace_2ubksp( \
```

File: keyboards/merge/um70/um70.h [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    { L30, L31, L32, L33, L34, L35, XXX, XXX }, \
+    { L40, L41, L42, L43, L44, L45, L46, L47 }, \
+    { R00, R01, R02, R03, R04, R05, XXX, R07 }, \
+    { R10, R11, R12, R13, R14, R15, R16, R17 }, \
+    { R20, R21, R22, R23, R24, R25, R26, R27 }, \
+    { R30, R31, R32, R33, R34, R35, R36, R37 }, \
+    { R40, R41, R42, R43, R44, R45, XXX, R47 } \
+}
+
+#define LAYOUT_LSpace_2UBksp( \
```
Comment (drashna) [inline keyboards/merge/um70/um70.h:61]:
```suggestion
#define LAYOUT_lspace_2ubksp( \
```

File: keyboards/merge/um70/um70.h [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    { L30, L31, L32, L33, L34, L35, XXX, XXX }, \
+    { L40, L41, L42, L43, L44, L45, L46, L47 }, \
+    { R00, R01, R02, R03, R04, R05, XXX, R07 }, \
+    { R10, R11, R12, R13, R14, R15, R16, R17 }, \
+    { R20, R21, R22, R23, R24, R25, R26, R27 }, \
+    { R30, R31, R32, R33, R34, R35, R36, R37 }, \
+    { R40, R41, R42, R43, R44, R45, XXX, R47 } \
+}
+
+#define LAYOUT_LSpace_SplitBksp( \
```
Comment (drashna) [inline keyboards/merge/um70/um70.h:80]:
```suggestion
#define LAYOUT_lspace_splitbksp( \
```

File: keyboards/merge/um70/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+ *   .---.  |-------------------------.          '---------------------------------|        |---|
+ *   |M0 |  |CapsL |  A|  S|  D|  F|  G|          |  H|  J|  K|  L|  ;|  :|   Retn |        |End|
+ *   |---|  |----------------------------.        '--------------------------------|  .---. '---'
+ *   |M1 |  |Shft    |  Z|  X|  C|  V|  B|          |  N|  M|  ,|  ,|  /|  Shift|     |Up | 
+ *   |---|  |----------------------------|          |---------------------------' .-----------.
+ *   |M2 |  |Ctl |Gui |Alt |Fn0 |Space   |          |    Space|   Alt|   Ctl|     |Lef|Dow|Rig|
+ *   '---'  '----------------------------'          '-----------------------'     '-----------'
+ */
+
+[_BASE] = LAYOUT_RSpace_SplitBksp(
```
Comment (drashna) [inline keyboards/merge/um70/keymaps/via/keymap.c:43]:
```suggestion
[_BASE] = LAYOUT_rspace_splitbksp(
```

File: keyboards/merge/um70/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+),
+
+[_ONE] = LAYOUT_RSpace_SplitBksp(
+                KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,    KC_TRNS,
+                KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                     KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS,   KC_TRNS,
+    KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                     KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,    KC_TRNS,         KC_TRNS,
+    KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                     KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                   KC_TRNS,
+    KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                              KC_TRNS, KC_TRNS, KC_TRNS,                               KC_TRNS, KC_TRNS, KC_TRNS
+),
+[_TWO] = LAYOUT_RSpace_SplitBksp(
```
Comment (drashna) [inline keyboards/merge/um70/keymaps/via/keymap.c:58]:
```suggestion
[_TWO] = LAYOUT_rspace_splitbksp(
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/merge/um70/um70.c [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+    { 224, 0  }, { 224, 13 }, { 224, 26 }, { 148, 39 }, { 185, 52 }, {  224,  64 },
+    
+    {   0,  0  }, {  32,  0 }, {  64, 0  }, {  96, 0  }, { 128, 0  }, { 160, 0  }, {  192,  0  }, {  224,  0  },
+    {   0,  16 }, {  32, 16 }, {  64, 16 }, {  96, 16 }, { 128, 16 }, { 160, 16 }, {  192,  16 }, {  223,  16  }, {  224,  16  },
+    {   0,  32 }, {  32, 32 }, {  64, 32 }, {  96, 32 }, { 128, 32 }, { 160, 32 }, {  192,  32 }, {  223,  32  },
+    {   0,  48 }, {  32, 48 }, {  64, 48 }, {  96, 48 }, { 128, 48 }, { 160, 48 }, {  192,  48 },
+    {   0,  46 }, {  32, 46 }, {  64, 64 }, {  96, 64 }, { 128, 64 }, { 160, 64 }
+
+}, {
+    2, 2, 2, 2, 2, 2,
```
[the remark this answers is not in the document]
Comment (duoshock) [reply keyboards/merge/um70/um70.c:62]: Thanks. It's updated.

**Comment by duoshock** [comment]
Thanks for reviewing. I've updated layout names and RGB matrix flag.

File: keyboards/merge/um70/config.h [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+#define MASTER_LEFT
+#define SPLIT_MODS_ENABLE
+
+#define RGB_DI_PIN D3
+#define RGBLIGHT_SPLIT
+#define RGBLED_NUM 83
+#define RGBLED_SPLIT { 39, 44 }
+#ifdef RGB_MATRIX_ENABLE
+#    define DRIVER_LED_TOTAL RGBLED_NUM
+#    define RGB_MATRIX_SPLIT RGBLED_SPLIT
```
Comment (drashna) [inline keyboards/merge/um70/config.h:46]: This may not work.

**Review by drashna — APPROVED** [review]

File: keyboards/merge/um70/rules.mk [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+ENCODER_ENABLE = yes
+SPLIT_KEYBOARD = yes
+OLED_DRIVER_ENABLE = yes
+BOOTMAGIC_ENABLE = lite
```
Comment (fauxpark) [inline keyboards/merge/um70/rules.mk:25]:
```suggestion
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
ENCODER_ENABLE = yes
SPLIT_KEYBOARD = yes
OLED_DRIVER_ENABLE = yes
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**